haskell-src/Concordium/Types/ProtocolVersion.hs

```diff
@@ -333,6 +336,13 @@ $( singletons
         supportsFlexibleCooldown AccountV3 = True
         supportsFlexibleCooldown AccountV4 = True
 
+        supportsAccountSuspension :: AccountVersion -> Bool
+        supportsAccountSuspension AccountV0 = False
+        supportsAccountSuspension AccountV1 = False
+        supportsAccountSuspension AccountV2 = False
+        supportsAccountSuspension AccountV3 = False
+        supportsAccountSuspension AccountV4 = True
```

Comment:
I defined the guard via the account version, because what really matters is whether the account has the suspend/resume flag or not. Could be wrong.